homeassistant/components/bluetooth/passive_update_processor.py

```diff
@@ -90,23 +282,49 @@ def __init__(
         self._processors: list[PassiveBluetoothDataProcessor] = []
         self._update_method = update_method
         self.last_update_success = True
+        self.restore_data: dict[str, RestoredPassiveBluetoothDataUpdate] = {}
+        self.restore_key = None
+        if config_entry := config_entries.current_entry.get():
+            self.restore_key = config_entry.entry_id
+        self._on_stop.append(async_register_coordinator_for_restore(self.hass, self))
```

Comment:
Should this be done in `async_register_processor` ?

Reply:
Since the coordinator can have multiple processors (one per platform) I did it here